<launch>
	<arg name="robot_name" default="sphero"/>       <!-- Name of the robot used for namespaces. -->
	<arg name="num_of_robots" default="6"/>         <!-- Number of robots used. -->
	<arg name="ctrl_loop_freq" default="10"/>       <!-- Frequency used by controllers. -->
  <arg name="data_stream_freq" default="100"/>    <!-- Position streaming frequency, used by Kalman filter. -->
	<arg name="debug_kalman" default="false"/>      <!-- Enable debugging for Kalman filter node. -->
	<arg name="associate_radius" default="0.05"/>   <!-- Radius for data association. -->
	<arg name="data_associated" default="false"/>		<!-- Is the data already associated? -->

	<!-- Set arguments as ros parameter so all nodes can access them. -->
	<param name="robot_name" type="string" value="$(arg robot_name)"/>
	<param name="num_of_robots" type="int" value="$(arg num_of_robots)"/>
	<param name="ctrl_loop_freq" type="int" value="$(arg ctrl_loop_freq)"/>
	<param name="data_stream_freq" type="int" value="$(arg data_stream_freq)"/>
	<param name="debug_kalman" type="bool" value="$(arg debug_kalman)"/>
	<param name="associate_radius" type="double" value="$(arg associate_radius)"/>
	<param name="data_associated" type="bool" value="$(arg data_associated)"/>

	<!-- Start Optitrack data streaming node. -->
	<include file="$(find mocap_optitrack)/launch/mocap.launch"/>

	<!-- Start Initialization node: Get initial Spheros positions. -->
	<node pkg="sphero_ros_control" type="sphero_init.py" name="initialization" output="screen"/>

	<!-- Start Kalman filter nodes launcher. -->
	<node pkg="sphero_localization" type="kalman_filter_launch.sh" name="kalman_launcher" output="screen"/>
</launch>
